services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -433,24 +436,77 @@ export class WorkflowService {
       projectIds,
     );
 
-    const totalWorkflowsCount = await this.workflowRuntimeDataRepository.count(
-      {
-        where: query.where,
-      },
-      projectIds,
+    const { workflowDefinitionIds, statuses, assigneeIds, includeUnassigned } = {
+      workflowDefinitionIds:
+        query.where.workflowDefinitionId &&
+        typeof query.where.workflowDefinitionId === 'object' &&
+        'in' in query.where.workflowDefinitionId &&
+        Array.isArray(query.where.workflowDefinitionId.in)
+          ? query.where.workflowDefinitionId.in
+          : [],
+      statuses:
+        query?.where?.status &&
+        typeof query.where.status === 'object' &&
+        'in' in query.where.status &&
+        Array.isArray(query.where.status.in)
+          ? query.where.status?.in
+          : [],
+      assigneeIds: filters?.assigneeId?.filter((id): id is string => id !== null) ?? [],
+      includeUnassigned: filters?.assigneeId?.includes(null),
+    };
+
+    const assigneeIdsParam = assigneeIds.length
+      ? Prisma.join(assigneeIds.map(id => Prisma.sql`${id}`))
+      : Prisma.sql``;
+
+    const workflowDefinitionIdsParam = Prisma.join(
+      workflowDefinitionIds.map(id => Prisma.sql`${id}`),
     );
 
-    if (page.number > 1 && totalWorkflowsCount < (page.number - 1) * page.size + 1) {
+    const statusesParam = statuses.length
+      ? Prisma.join(statuses.map(status => Prisma.sql`${status}`))
+      : Prisma.sql``;
+
+    const projectIdsParam = Prisma.join((projectIds || []).map(id => Prisma.sql`${id}`));
```

Comment:
The default to an empty array here doesn't seem right, we should probably fail this action when there is no project filter

Reply:
As discussed, it is safe